futurex_openedx_extensions/helpers/tasks_utils.py

```diff
@@ -0,0 +1,198 @@
+from urllib.parse import urlencode
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.files.base import ContentFile
+from django.core.files.storage import default_storage
+from django.urls import resolve
+from rest_framework.request import Request
+from rest_framework.test import APIRequestFactory
+
+from futurex_openedx_extensions.helpers.exceptions import FXCodedException, FXExceptionCodes
+from futurex_openedx_extensions.helpers.models import DataExportTask
+
+User = get_user_model()
+
+
+def _get_user(user_id: Optional[int]) -> Any:
+    """Get User from user_id"""
+    if user_id and isinstance(user_id, int):
+        return User.objects.get(id=user_id)
+    raise FXCodedException(
+        code=FXExceptionCodes.USER_NOT_FOUND,
+        message=f'CSV Export: Invalid user id: {user_id}',
+    )
+
+
+def _get_view_class_instance(path: str) -> Any:
+    """Create view class instance"""
+    if path:
+        view_func = resolve(path)
+        return view_func.func
+    raise FXCodedException(
+        code=FXExceptionCodes.EXPORT_CSV_MISSING_REQUIRED_PARAMS,
+        message=f'CSV Export: Missing required params "path" {path}',
+    )
+
+
+def _get_mocked_request(url_with_query_str: str, fx_info: dict) -> Request:
+    """Create mocked request"""
+    factory = APIRequestFactory()
+    mocked_request = factory.get(url_with_query_str)
+    mocked_request.user = fx_info['user']
+    mocked_request.fx_permission_info = fx_info
+    return mocked_request
+
+
+def _get_response_data(response: Any) -> Tuple:
+    """Get response data"""
+    if response.status_code != 200:
+        raise FXCodedException(
+            code=FXExceptionCodes.EXPORT_CSV_VIEW_RESPONSE_FAILURE,
+            message=f'CSV Export: View returned status code: {response.status_code}',
+        )
+    if not response.data:
+        raise FXCodedException(
+            code=FXExceptionCodes.EXPORT_CSV_VIEW_RESPONSE_FAILURE,
+            message='CSV Export: Unable to process view response.',
+        )
+    data = response.data.get('results')
+    if data is None or not isinstance(data, list):
+        raise FXCodedException(
+            code=FXExceptionCodes.EXPORT_CSV_VIEW_RESPONSE_FAILURE,
+            message='CSV Export: The "results" key is missing or is not a list.',
+        )
+
+    count = response.data.get('count')
+    if count is None or not isinstance(count, int):
+        raise FXCodedException(
+            code=FXExceptionCodes.EXPORT_CSV_VIEW_RESPONSE_FAILURE,
+            message='CSV Export: The "count" key is missing or is not an int.',
+        )
+    return data, count
+
+
+def _paginated_response_generator(
+    fx_info: dict, view_data: dict, view_instance: Any
+) -> Generator:
+    """Generator to yield paginated responses."""
+    url = view_data['url']
+    kwargs = view_data.get('kwargs', {})
+    processed_records = 0
+    progress = 0
+    while url:
```

Comment:
I don't see that necessary. I couldn't think of a scenario that could make it loop forever

Reply:
I think timeout will be good to add.

Reply:
Thanks @tehreem-sadat and Shadi, not to contradict myself but timeout failure handling is its own logic and needs to be tested.

I think it makes sense to add in a future work, rather than immediately. Let me know your preference.